addons/web/static/tests/core/domain_tests.js

```diff
@@ -59,6 +64,34 @@ QUnit.module('core', {}, function () {
         assert.notOk(new Domain(0).compute({}));
     });
 
+    QUnit.test("invalid domains should not succeed", function (assert) {
+        assert.expect(6);
+        assert.throws(
+            () => new Domain(['|', ['hr_presence_state', '=', 'absent']]),
+            /invalid domain .* \(missing 1 segment/
+        );
+        assert.throws(
+            () => new Domain(['|', '|', ['hr_presence_state', '=', 'absent'], ['attendance_state', '=', 'checked_in']]),
+            /invalid domain .* \(missing 1 segment/
+        );
+        assert.throws(
+            () => new Domain(['|', ['usps_service', '!=', 'First Class']]),
+            /invalid domain .* \(missing 1 segment/
+        );
+        assert.throws(
+            () => new Domain(['|', ['composition_mode', '!=', 'mass_post']]),
+            /invalid domain .* \(missing 1 segment/
+        );
+        assert.throws(
+            () => new Domain(['&', ['usps_service', '!=', 'First Class']]),
+            /invalid domain .* \(missing 1 segment/
+        );
+        assert.throws(
+            () => new Domain(['&', ['composition_mode', '!=', 'mass_post']]),
+            /invalid domain .* \(missing 1 segment/
```

Comment:
some tests are redundant

Reply:
Yeah originally I was evaluating them so they would have different contexts, but then the validation moved to parsing and the cases became redundant-er.